lib/middleware.js

```diff
@@ -6,14 +6,23 @@ const bunyan = require('bunyan');
 const serializers = require('./serializers');
 
 function child(fields) {
-  const newContext = _.cloneDeep(this);
+  const refProps = this.refProps;
+  const newContext = _.cloneDeepWith(this, obj => {
```

Comment:
According to [the docs](https://lodash.com/docs/4.17.4#cloneWith), customizer is called with `value [, index|key, object, stack`. You do not need to `mapValues`, just check the key. Like:
```js
const newContext = _.cloneDeepWith(this, (value, key) => {
  return _.includes(refProps, key)
    ? value
    : undefined // to trigger default behaviour
});
```

Reply:
I swear I tried this and was getting an error, but must have been something else because it works now